pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java

```diff
@@ -1690,6 +1700,23 @@ public void testCompilationInvokedFunction() {
     greaterThan = pinotQuery.getFilterExpression().getFunctionCall();
     nowTs = greaterThan.getOperands().get(1).getLiteral().getLongValue();
     Assert.assertEquals(nowTs, 1577836800000L);
+
+    query = "SELECT ago('PT1H') FROM foo";
+    lowerBound = System.currentTimeMillis() - ONE_HOUR_IN_MS;
+    pinotQuery = CalciteSqlParser.compileToPinotQuery(query);
+    nowTs = pinotQuery.getSelectList().get(0).getLiteral().getLongValue();
+    upperBound = System.currentTimeMillis() - ONE_HOUR_IN_MS;
+    Assert.assertTrue(nowTs >= lowerBound);
+    Assert.assertTrue(nowTs <= upperBound);
+
+    query = "SELECT a FROM foo where time_col > ago('PT1H')";
+    lowerBound = System.currentTimeMillis() - ONE_HOUR_IN_MS;
```

Comment:
Can we also ensure that query gets rewritten ? I think the query gets rewritten to ?

`SELECT a FROM foo where time_col - ago('PT1H') > 0`

Reply:
Also, since ago(PT1H) can be invoked at compile time so the value will get substituted right ?

Reply:
Yes. L1717  verified that ago(PT1H) is invoked at compile time and the value will get substituted
```
greaterThan = pinotQuery.getFilterExpression().getFunctionCall();
    nowTs = greaterThan.getOperands().get(1).getLiteral().getLongValue();
```

Reply:
> Can we also ensure that query gets rewritten ? I think the query gets rewritten to ?
```
SELECT a FROM foo where time_col - ago('PT1H') > 0
```
I don't think the query get rewritten, If I output the complied query, it looks like:
```
System.out.println(pinotQuery);
PinotQuery(dataSource:DataSource(tableName:foo), selectList:[Expression(type:IDENTIFIER, identifier:Identifier(name:a))], filterExpression:Expression(type:FUNCTION, functionCall:Function(operator:GREATER_THAN, operands:[Expression(type:IDENTIFIER, identifier:Identifier(name:time_col)), Expression(type:LITERAL, literal:<Literal longValue:1622672158605>)])))

```

Reply:
Got it. Due to compile time value substitution, there is no need to rewrite.